controllers/clustermanagementaddon_controller.go

```diff
@@ -0,0 +1,590 @@
+					Port: 8091,
+				},
+			},
+		},
+	}
+
+	const signerSecretName = "proxy-server-ca"
+	proxyServerCASecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: config.Spec.ProxyServer.Namespace,
+			Name:      signerSecretName,
+		},
+		Data: map[string][]byte{
+			"ca.crt": c.SelfSigner.CAData(),
+		},
+	}
+	proxyServerDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: config.Spec.ProxyServer.Namespace,
+			Name:      config.Name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &config.Spec.ProxyServer.Replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					common.LabelKeyComponentName: common.ComponentNameProxyServer,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						common.LabelKeyComponentName: common.ComponentNameProxyServer,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: agentAuthServiceAccount.Name,
+					Containers: []corev1.Container{
+						{
+							Name:  common.ComponentNameProxyServer,
+							Image: config.Spec.ProxyServer.Image,
+							Args: []string{
+								"--server-count=" + strconv.Itoa(int(config.Spec.ProxyServer.Replicas)),
+								"--proxy-strategies=destHost",
+								"--server-ca-cert=/etc/server-ca-pki/ca.crt",
+								"--server-cert=/etc/server-pki/tls.crt",
+								"--server-key=/etc/server-pki/tls.key",
+								"--cluster-ca-cert=/etc/server-ca-pki/ca.crt",
+								"--cluster-cert=/etc/agent-pki/tls.crt",
+								"--cluster-key=/etc/agent-pki/tls.key",
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "proxy-server-ca-certs",
+									ReadOnly:  true,
+									MountPath: "/etc/server-ca-pki/",
+								},
+								{
+									Name:      "proxy-server-certs",
+									ReadOnly:  true,
+									MountPath: "/etc/server-pki/",
+								},
+								{
+									Name:      "proxy-agent-certs",
+									ReadOnly:  true,
+									MountPath: "/etc/agent-pki/",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "proxy-server-ca-certs",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: signerSecretName,
+								},
+							},
+						},
+						{
+							Name: "proxy-server-certs",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: config.Spec.Authentication.CertificateMounting.Secrets.SigningProxyServerSecretName,
+								},
+							},
+						},
+						{
+							Name: "proxy-agent-certs",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: config.Spec.Authentication.CertificateMounting.Secrets.SigningAgentServerSecretName,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	if err := c.Client.Create(context.TODO(), agentAuthServiceAccount); err != nil {
```

Comment:
We only create the resources, we may need to consider the update cases

Reply:
now it handles update